griptape/tokenizers/base_tokenizer.py

```diff
@@ -4,10 +4,22 @@
 from griptape import utils
 
 
-@define(frozen=True)
+@define()
 class BaseTokenizer(ABC):
+    MODEL_PREFIXES_TO_MAX_OUTPUT_TOKENS = {}
+    MODEL_PREFIXES_TO_MAX_TOKENS = {}
+
+    model: str = field(kw_only=True)
     stop_sequences: list[str] = field(default=Factory(lambda: [utils.constants.RESPONSE_STOP_SEQUENCE]), kw_only=True)
-    max_tokens: int = field(kw_only=True)
+    max_tokens: int = field(kw_only=True, default=None)
+    max_output_tokens: int = field(kw_only=True, default=None)
+
+    def __attrs_post_init__(self) -> None:
+        if self.max_tokens is None:
+            self.max_tokens = self._default_max_tokens()
+
+        if self.max_output_tokens is None:
+            self.max_output_tokens = self._default_max_output_tokens()
```

Comment:
Our usual method of handling attrs defaults that depend on each other does not work well here because of partially initialized objects.